packages/serverpod_test/lib/src/test_session_builder.dart

```diff
@@ -4,135 +4,115 @@ import 'test_serverpod.dart';
-  final List<InternalTestSession> _allTestSessions;
+class InternalTestSessionBuilder extends TestSessionBuilder {
+  final List<InternalServerpodSession> _allTestSessions;
   final TestServerpod _testServerpod;
-  AuthenticationOverride? _authenticationOverride;
+  final AuthenticationOverride? _authenticationOverride;
   final bool _enableLogging;
-
-  /// The underlying Serverpod session
-  @override
-  InternalServerpodSession serverpodSession;
-
-  @override
-  Database get db => serverpodSession.db;
-
-  @override
-  Future<AuthenticationInfo?> get authenticationInfo async {
-    return serverpodSession.authenticated;
-  }
-
-  @override
-  Transaction? get transaction =>
-      serverpodSession.transactionManager.currentTransaction;
+  final InternalServerpodSession _mainServerpodSession;
 
   /// Creates a new internal test session.
-  InternalTestSession(
+  InternalTestSessionBuilder(
     TestServerpod testServerpod, {
     AuthenticationOverride? authenticationOverride,
-    InternalTestSession? sessionWithDatabaseConnection,
+    InternalTestSessionBuilder? sessionWithDatabaseConnection,
     required bool enableLogging,
-    required List<InternalTestSession> allTestSessions,
-    required this.serverpodSession,
+    required List<InternalServerpodSession> allTestSessions,
+    required InternalServerpodSession mainServerpodSession,
   })  : _allTestSessions = allTestSessions,
         _authenticationOverride = authenticationOverride,
         _testServerpod = testServerpod,
-        _enableLogging = enableLogging {
-    _allTestSessions.add(this);
-    _configureServerpodSession(serverpodSession);
+        _enableLogging = enableLogging,
+        _mainServerpodSession = mainServerpodSession;
+
+  @override
+  Session build() {
+    return internalBuild();
   }
 
   @override
-  TestSession copyWith({
+  TestSessionBuilder copyWith({
     AuthenticationOverride? authentication,
     bool? enableLogging,
-    String endpoint = '',
-    String method = '',
   }) {
-    var newServerpodSession = _testServerpod.createSession(
-      enableLogging: enableLogging ?? _enableLogging,
-      endpoint: endpoint,
-      method: method,
-      rollbackDatabase: serverpodSession.rollbackDatabase,
-      transactionManager: serverpodSession.transactionManager,
-    );
-
-    return InternalTestSession(
+    return InternalTestSessionBuilder(
       _testServerpod,
       allTestSessions: _allTestSessions,
       authenticationOverride: authentication ?? _authenticationOverride,
       enableLogging: enableLogging ?? _enableLogging,
-      serverpodSession: newServerpodSession,
+      mainServerpodSession: _mainServerpodSession,
     );
   }
 
+  /// Creates a new Serverpod session with the specified [endpoint] and [method].
+  /// Should only be used by generated code.
+  Session internalBuild({
+    String endpoint = '<not applicable: created by user in test>',
+    String method = '<not applicable: created by user in test>',
```

Comment:
I thought this could be handy for debugging purposes if for some reason someone had some logic around this.